lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -320,7 +327,13 @@ def self.create_ae_attrs(attrs, name, vmdb_object, objects = [MiqServer.my_serve
     array_objects.each do |array_object|
       # Each array attribute is tagged with Array:: before the attribute key unless it already starts with Array::
       array_attr_key = array_object
-      if !array_object.starts_with?("Array::")
+      if !ae_attrs[array_object].empty? && ae_attrs[array_object].first.split("::").first == "Classification"
```

Comment:
New line 326 hurts my brain but let's not change that right now...

326 guarantees this first conditional is an array, so the empty check is fine, but, the next one is a bit concerning.

Maybe something like this:

```ruby
if !ae_attrs[array_object].empty? && ae_attrs[array_object].first.to_s.split("::").first == "Classification"
```